### PR TITLE
vscode: report LSP panic as telemetry error

### PR DESCRIPTION
**Pull request by ogoffart**
We write the panic message in a file to communicate with the vscode extension

### CODE REVIEW

**Comment by tronical** [comment]
Looks good to me, please also extend `telemetry.json`

**Comment by ogoffart** [comment]
> please also extend telemetry.json

I did something, but i have no idea what's supposed to be there.

File: editors/vscode/telemetry.json [hunk 1]
```diff
@@ -1,6 +1,23 @@
 {
     "events": {
         "extension-activated": {}
+        "lsp-panic": {
```
Comment (tronical) [inline editors/vscode/telemetry.json:4]: Needs comma at end of previous line

File: editors/vscode/telemetry.json [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+            "version": {
+                "classification": "PublicNonPersonalData",
+                "purpose": "PerformanceAndHealth",
+                "comment": "The version of the Slint LSP server"
+            }
```
Comment (tronical) [inline editors/vscode/telemetry.json:9]:
```suggestion
            },
```

File: editors/vscode/telemetry.json [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
+            "message": {
+                "classification": "CallstackOrException",
+                "purpose": "PerformanceAndHealth",
+                "comment": "The message of the panic"
+            }
```
Comment (tronical) [inline editors/vscode/telemetry.json:14]:
```suggestion
            },
```

**Review by tronical — COMMENTED** [review]
Contents looks good, but needs to be valid JSON

**Pull request merged**